### PR TITLE
[[provide]] node in build plan

### PR DESCRIPTION
**Pull request by jabrown85**
Ensure that the buildpack defines what it provides so that other buildpacks that depend on this buildpack can specify that requirement. The reason `node` was chosen was to align with the name that the riff node buildpack uses to require `node`.

https://github.com/buildpacks/spec/blob/master/buildpack.md#process
https://github.com/projectriff/node-function-buildpack/blob/07eacdbe9fe49b8ce8f726fc617a5bc895b9a7b4/node/buildpack.go#L72

### CODE REVIEW

File: bin/detect [hunk 1]
```diff
@@ -5,10 +5,16 @@ set -eo pipefail
...
 # shellcheck source=/dev/null
 source "$bp_dir/lib/detect.sh"
 
 if ! detect_package_json "$build_dir" ; then
   exit 100
+else
+  cat << EOF > "$build_plan"
+  [[provides]]
+  name = "node"
+EOF
```
Comment (danielleadams) [inline bin/detect]: Can we do the same thing here as requested in the npm buildpack: https://github.com/heroku/nodejs-npm-buildpack/pull/18#discussion_r372547386

Then we can add a unit test for it.

**Comment by danielleadams** [comment]
@jabrown85 Do you mind updating the Changelog for this too?

**Comment by jabrown85** [comment]
@danielleadams I bumped the version, updated changelog, and added a test. I also tested that this buildpack works with the builders as-is. Let me know if you want me to test anything else. I'll follow up on the npm-modules one next.

**Comment by danielleadams** [comment]
Looks good - thanks for the changes! I'm going to go ahead and merge this, but I'm going to add one more change to this release before I tag it.

**Pull request merged**